operations/host_provision.go

```diff
@@ -87,6 +113,29 @@ func hostProvision() cli.Command {
 	}
 }
 
+func postHostIsUp(ctx context.Context, comm client.Communicator, hostID, hostSecret, cloudProvider string) (*restmodel.APIHost, error) {
+	var ec2InstanceID string
+	if cloud.IsEC2InstanceID(hostID) {
+		ec2InstanceID = hostID
+	} else if cloudProvider != "" && evergreen.IsEc2Provider(cloudProvider) {
+		var err error
+		ec2InstanceID, err = agentutil.GetEC2InstanceID(ctx)
+		if err != nil {
```

Comment:
Continue on error is necessary here for monitoring purposes. We can't return in this case because if we do, the host fails to provision completely and we have no way to know that it failed because it silently exited here. If it pings the app server, we at the very least can track that it did come up, even without an EC2 instance ID, and monitor/address if it becomes an issue. The assumption is that the host will almost always give the EC2 instance ID, but if that somehow turns out to not be the case in the future (maybe EC2 changes/removes the metadata endpoint), we would want to be able to monitor that something is wrong.